strutil/strutil.go

```diff
@@ -380,3 +380,25 @@ func JoinNonEmpty(strs []string, sep string) string {
 	}
 	return strings.Join(nonEmpty, sep)
 }
+
+// CommonPrefix returns the prefix common to a slice of strings.
+//
+// The prefix for an empty slice is an empty string.
+// The prefix for a slice with one element is the element itself.
+func CommonPrefix(elems []string) string {
+	if len(elems) == 0 {
+		return ""
+	}
+
+	prefix := elems[0]
+	for _, s := range elems {
```

Comment:
looks good. One tiny nitpick: we can skip the first element
```suggestion
	for _, s := range elems[1:] {
```

Reply:
Ah, good point. Thanks!

Reply:
Done